.github/workflows/ci.yml

```diff
@@ -45,4 +110,251 @@ jobs:
+        with:
+          ref: ${{ github.event.workflow_run.head_sha }}
+      - name: Fetch tags
+        run: git fetch --prune --unshallow --tags
+      - name: Get info
+        id: info
+        env:
+          BRANCH: ${{ github.event.workflow_run.head_branch }}
+        # The tag name should be retrieved lazily, as tagging may be delayed.
+        run: |
+          echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+          if [[ "$BRANCH" = "release" ]]; then
+            TAG=$(git tag --points-at HEAD)
+            if [[ ! -z "$TAG" ]]; then
+              echo "::set-output name=tag::$TAG"
+              echo "::set-output name=tag_short::${TAG#v}"
+            else
+              echo "::set-output name=name::rc"
+            fi
+          else
+            echo "::set-output name=name::nightly"
+          fi
+      - name: Show info
+        env:
+          SHA_SHORT: ${{ steps.info.outputs.sha_short }}
+          TAG: ${{ steps.info.outputs.tag }}
+          TAG_SHORT: ${{ steps.info.outputs.tag_short }}
+          NAME: ${{ steps.info.outputs.name }}
+        run: echo "sha_short=$SHA_SHORT, tag=$TAG, tag_short=$TAG_SHORT, name=$NAME"
+
+  build-server:
+    name: Build and release server
+    runs-on: ubuntu-latest
+    needs: [ci-web, ci-server, info]
+    if: ${{ github.event_name == 'push' && contains(needs.ci-server.result, 'success') && !contains(needs.ci-web.result, 'failure') && !contains(needs.ci-web.result, 'cancelled') && (needs.info.outputs.name || needs.info.outputs.tag) }}
+    env:
+      ARTIFACTS: server/dist/reearth_*.*
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.workflow_run.head_sha }}
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.19"
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          args: release --rm-dist ${{ env.SNAPSHOT }}
+          workdir: server
+        env:
+          SNAPSHOT: ${{ !needs.info.outputs.tag && '--snapshot' || '' }}
+          GORELEASER_CURRENT_TAG: ${{ needs.info.outputs.tag || '0.0.0' }}
+      - name: Rename artifacts
+        if: needs.info.outputs.name
+        run: for f in $ARTIFACTS; do mv $f $(echo $f | sed -E 's/_0\.0\.0-SNAPSHOT-[^_]*/_${{ needs.info.outputs.name }}/'); done
+      - name: List artifacts
+        run: ls -l server/dist
+      - name: Release nightly/rc
+        if: needs.info.outputs.name
+        uses: ncipollo/release-action@v1
+        with:
+          allowUpdates: true
+          artifacts: ${{ env.ARTIFACTS }}
+          commit: ${{ github.sha }}
+          name: ${{ needs.info.outputs.name }}
+          tag: ${{ needs.info.outputs.name }}
+          body: ${{ needs.info.outputs.sha_short }}
+          prerelease: true
+
+  web-release:
+    name: Release web
+    runs-on: ubuntu-latest
+    needs: [ci-web, ci-server, info]
```

Comment:
```suggestion
    needs: info
```

is not working?

Reply:
Similar to https://github.com/reearth/reearth/pull/393/files#r1169905369